docs/theming.md

```diff
@@ -0,0 +1,299 @@
+| ------------------------------------------ | ---------------------------------------------------------- |
+| `--mdc-theme-text-primary-on-dark`        | Primary text on top of a dark-colored background.         |
+| `--mdc-theme-text-secondary-on-dark`      | Secondary text on top of a dark-colored background.       |
+| `--mdc-theme-text-hint-on-dark`           | Hint text on top of a dark-colored background.            |
+| `--mdc-theme-text-disabled-on-dark`       | Disabled text on top of a dark-colored background.        |
+| `--mdc-theme-text-icon-on-dark`           | Icons on top of a dark-colored background.                |
+
+
+## Typography
+
+Material Theming defines a hierarchy of type styles, from headlines to body text and captions. 
+The default Material Design theme uses the Roboto font for all of these styles.
+
+The easiest way to customize your application is to set a new default font family. If necessary, you
+can also set individual properties for each of the named type styles.
+
+See also: [Material design type system](https://material.io/design/typography/the-type-system.html#)
+
+### Typography properties
+
+MWC provides a set of CSS custom properties to define typography for your application. 
+
+The `--mdc-typeography-font-family` property sets the default font family for all components.
+
+For more granular control, you can set individual type properties for each of the styles 
+used in the type system:
+
+Style | Description
+--- | ---
+`headline1` | The largest text on the screen, reserved for short, important text or numerals
+`headline2` | Headline variant 2
+`headline3` | Headline variant 3
+`headline4` | Headline variant 4
+`headline5` | Headline variant 5
+`headline6` | Headline variant 6
+`subtitle1` | Smaller than headline, reserved for medium-emphasis text that is shorter in length
+`subtitle2` | Subtitle variant 2
+`body1` | Used for long-form writing
+`body2` | Body variant 2
+`caption` | Used sparingly to annotate imagery
+`button` | A call to action used by different types of buttons
+`overline` | Used sparingly to introduce a headline
+
+For a given style, MWC supports the following custom properties (where `<STYLE>` is one of the style names, above):
+
+CSS custom property | Description
+--- | ---
+`--mdc-typography-font-family` | The base font-family
+`--mdc-typography-<STYLE>-font-family` | The font-family for `<STYLE>`. 
+`--mdc-typography-<STYLE>-font-size` | The font-size for `<STYLE>`. 
+`--mdc-typography-<STYLE>-line-height` | The line-height for `<STYLE>`.  
+`--mdc-typography-<STYLE>-font-weight` | The font-weight for `<STYLE>`. 
+`--mdc-typography-<STYLE>-letter-spacing` | The letter-spacing for `<STYLE>`. 
+`--mdc-typography-<STYLE>-text-decoration` | The text-decoration for `<STYLE>`. 
+`--mdc-typography-<STYLE>-text-transform` | The text-transform for `<STYLE>`. 
+
+For example, to set typeography properties for a button, you might use the following CSS:
```

Comment:
Hmm... Yeah, I was looking for a way to clarify the format. Maybe I'll just add a one-sentence "For example, to set font-size for your main body text, use `--mdc-typography-body1-font-size`."

Additional thought: should we have an example of _applying_ one of these? For example, I'm not sure whether MWC actually uses `body1` anywhere.

Reply:
That's a tricky question. Right now none of these custom properties are technically defined. If I wrote this:

```css
p {
  font-family: var(--mdc-typography-font-family);
}
```
Nothing would happen. Each component that uses it defines a fallback. So there's really no way for the user to "consume" the default fallback values.

For typography specifically, this will change when we introduce mwc-typography, which will let them apply classes to style things by default.

Reply:
Got it. But that will only apply to MDC, right? Applying classes isn't going to work for people using MWC.

Reply:
It'll be for MWC as well. The typography package will export CSSResults. They can either be added directly to LitElement styles and the classes used within an element's shadow dom, or they can be added to a document stylesheet to make the classes available globally.

Reply:
👍

Reply:
OK, cool. I'll mark this resolved for now, and we can add details on using the typography package when it's available.